app/controllers/nsm/imports_controller.rb

```diff
@@ -30,13 +21,19 @@ def create
             handle_validation_errors
           end
         end
+      else
+        render :new
       end
-
-      render :new
     end
 
     def errors
-      @errors = JSON.parse(errors_file_path.read)
+      begin
+        @errors = JSON.parse(errors_file_path.read)
+      rescue StandardError
```

Comment:
That's not technically the only thing that can go wrong here. Should only return the error if the file does indeed not exist.

Reply:
rectified